connectivity/suite.go

```diff
@@ -58,19 +63,35 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 	)
 
 	// Test with an allow-all-except-world (and unmanaged) policy.
-	ct.NewTest("allow-all-except-world").WithPolicy(allowAllExceptWorldPolicyYAML).
-		WithScenarios(
-			tests.PodToPod(""),
-			tests.ClientToClient(""),
-			tests.PodToService(""),
-			// We are skipping the following checks because NodePort is
-			// intended to be used for N-S traffic, which conflicts with
-			// policies. See GH-17144.
-			// tests.PodToRemoteNodePort(""),
-			// tests.PodToLocalNodePort(""),
-			tests.PodToHost(""),
-			tests.PodToExternalWorkload(""),
-		)
+	if strings.HasPrefix(ct.FetchCiliumPodImageTag(), "v1.11") {
```

Comment:
I think this will not work e.g. for CI images in `cilium/cilium`, where the tag will be some hash. I assume we'd mistakenly use the pre 1.11 version in such cases.

To fix this, I'd suggest to add a `--base-version` CLI arg for `cilium connectivity test` (similar to the one for `cilium install`) which will allow to set the assumed base version for such cases in CI. Alternatively we could just fall back to `defaults.Version` in case the image tag is not a semver version.